### PR TITLE
Improve module and cli speed using cdnjs.com's official API

### PR DESCRIPTION
**Pull request by alexjab**
https://github.com/phuu/cdnjs/issues/14

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules/
```
Comment (tgvashworth) [inline .gitignore:1]: I'd recommend adding these to a global `.gitignore` — would rather keep 'em out of the repo.

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 language: node_js
 node_js:
-  - "0.10"
-  - "0.9"
\ No newline at end of file
+  - "0.10.33"
+before_script:
+  - "npm install -g jshint mocha"
```
Comment (tgvashworth) [inline .travis.yml:5]: These can go in `devDependencies` — npm will copy these modules' executables into `node_modules/.bin`, and will look there when running scripts.

File: README.md [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
+Search and URL retrieval from cdnjs
+
+![terminal](http://i.imgur.com/QJ0gnLT.gif)
+
 [![build status](https://secure.travis-ci.org/phuu/cdnjs.png)](http://travis-ci.org/phuu/cdnjs)
 
-A search and URL retrieval tool for [cdnjs](//cdnjs.com). It can be used globally, on your command line, or as a module.
+A search and URL retrieval tool for [cdnjs](//cdnjs.com). It can be used globally on your command line, or as a module.
+
+Its version 3.2.0 powers the [pulldown-api](https://github.com/phuu/pulldown-api).
```
Comment (tgvashworth) [inline README.md:11]: Don't worry about the version, we'll upgrade.

**Comment by tgvashworth** [comment]
Thanks for this @alexjab — sorry for the delay.

I left some minor comments on the first couple of files and I may take another run at it at some point, but I'm afraid it's going to be tough.

It's never a nice thing to say, but I'm unlikely to merge in this as the diff is just too big. It's really hard for me to get an understanding the extent of the changes as it's a basically rewrite and so I can't feel confident releasing it, or be sure of it's maintainability in future. In that sense, you might be better off publishing this as its own module.

**Comment by alexjab** [comment]
Hi,

I concur, the changes are quite significant; There may be another way around rewriting everything as I did, but the easiest way would be indeed to create my own new module.

I see no reason not to close the PR, but can you leave the issue thread open until I release my version on npm, in order to reference it there ?

BR,

Alex

**Pull request closed without merging**